File: public/codesamples/posts/graphes_grids/code3/NodeMap.java
package posts.graphes_grids.code3;

import java.util.HashMap;
import java.util.List;
import java.util.ArrayList;
import java.util.Map;

public class NodeMap {
  private Map<Integer, HashMap<Integer, Node>> nodes = new HashMap<>();
  private int[][] neighbours = new int[][] { { 0, 1 }, { 0, -1 }, { 1, 0 }, { -1, 0 } };

  // Haritayı oluşturan fonksiyon
  public void createMap(int horizontalNodeCount, int verticalNodeCount) {
    for (int i = 0; i < verticalNodeCount; i++) {
      if (!nodes.keySet().contains(i)) {
        nodes.put(i, new HashMap<>());
      }
      for (int j = 0; j < horizontalNodeCount; j++) {
        nodes.get(i).put(j, new Node(j, i));
      }
    }
    for (Map.Entry<Integer, HashMap<Integer, Node>> i : nodes.entrySet()) {
      for (Map.Entry<Integer, Node> node : i.getValue().entrySet()) {
        for (Node neighbour : getNeighbour(node.getValue())) {
          node.getValue().getConnections().add(new Connection(node.getValue(), neighbour));
        }
      }
    }
  }

  // Düğümlerin komşularını veren fonksiyon
  public List<Node> getNeighbour(Node node) {
    List<Node> result = new ArrayList<>();
    for (int[] arr : neighbours) {
      if (nodes.keySet().contains(node.getY() + arr[1])) {
        if (nodes.get(node.getY() + arr[1]).keySet().contains(node.getX() + arr[0])) {
          result.add(nodes.get(node.getY() + arr[1]).get(node.getX() + arr[0]));
        }
      }
    }
    return result;
  }
}

File: public/codesamples/posts/graphes_grids/code3/NodeMap.py
from Node import Node;
from Connection import Connection;

nodes = dict()
neighbours = [
  [0, 1],
  [0, -1],
  [1, 0],
  [-1, 0],
]
horizontalNodeCount = 10
verticalNodeCount = 10

def getNeigboor(node):
  result = []
  for arr in neighbours:
    if nodes in arr[1]+node.y:
      if nodes[arr[1]+node.y] in arr[0]+node.x:
        result.append(nodes[arr[1]+node.y][arr[0]+node.x])
  return result

for i in range(verticalNodeCount):
  if not (nodes in i):
    nodes[i]=dict()
  for j in range(horizontalNodeCount):
    nodes[i][j]=Node(j,i)

for item in nodes.values():
  for node in item.values():
    for neigboor in getNeigboor(node):
      node.connections.add(Connection(node,neigboor))